api/ipfsproxy/config.go

```diff
@@ -50,16 +55,34 @@ type Config struct {
 	// Server-side amount of time a Keep-Alive connection will be
 	// kept idle before being reused
 	IdleTimeout time.Duration
+
+	// A list of custom headers that should be extracted from
+	// IPFS daemon responses and re-used in responses from hijacked paths.
+	// This is only useful if the user has configured custom headers
+	// in the IFPS daemon. CORS-related headers are already
+	// taken care of by the proxy.
+	ExtractHeadersExtra []string
+
+	// If the user wants to extract some extra custom headers configured
+	// on the IPFS daemon so that they are used in hijacked responses,
+	// this request path will be used. Defaults to /version. This will
+	// trigger a single request to extract those headers and remember them
```

Comment:
This functionality means that upgrades to the ipfs daemon (in deployment situations) _**may**_ require restarting the associated ipfs-cluster peer.